### PR TITLE
[SYCLomatic] Migration of cudaStreamCaptureStatus enum, cudaStreamIsCapturing API

### PR DESCRIPTION
**Pull request by daiyaan-ahmed6**
Signed-off-by: Daiyaan Ahmed <daiyaan.ahmed@intel.com>

### CODE REVIEW

File: clang/lib/DPCT/MapNames.cpp [hunk 1]
```diff
@@ -364,6 +364,11 @@ void MapNames::setExplicitNamespaceMap() {
       {"cusparseHandle_t", std::make_shared<TypeNameRule>(
                                getDpctNamespace() + "sparse::descriptor_ptr")},
       {"cudaMemoryAdvise", std::make_shared<TypeNameRule>("int")},
+      {"cudaStreamCaptureStatus",
```
Comment (ziranzha) [inline clang/lib/DPCT/MapNames.cpp:367]: When the extension graph isn't enabled, maybe we should emit a warning instead of migrating to `int` here.

Comment (daiyaan-ahmed6) [reply clang/lib/DPCT/MapNames.cpp:367]: Done.

File: clang/lib/DPCT/ASTTraversal.cpp [hunk 2]
```diff
@@ -3430,6 +3438,24 @@ void EnumConstantRule::runRule(const MatchFinder::MatchResult &Result) {
...
+    if (!DpctGlobalInfo::useExtGraph()) {
+      report(E->getBeginLoc(), Diagnostics::TRY_EXPERIMENTAL_FEATURE, false,
+             "cudaStreamCaptureStatusNone",
+             "--use-experimental-features=graph");
+      return;
+    }
+  } else if (EnumName == "cudaStreamCaptureStatusInvalidated") {
+    report(E->getBeginLoc(), Diagnostics::API_NOT_MIGRATED, false,
+           "cudaStreamCaptureStatusInvalidated");
+    return;
```
Comment (ziranzha) [inline clang/lib/DPCT/ASTTraversal.cpp]: Can we merge the three conditions?

Comment (daiyaan-ahmed6) [reply clang/lib/DPCT/ASTTraversal.cpp]: Done. Merged 1st & Second condition

**Review by ziranzha — APPROVED** [review]

**Review by intwanghao — APPROVED** [review]
LGTM

**Pull request merged**